R/readWav.R

```diff
@@ -8,26 +8,36 @@ readWav = function(filename, start = 1, end = 100, units = "minutes") {
   B = as.data.frame(B, stringsAsFactors = TRUE)
   #-------------------------------------------------------
   # extract info from header: fileEncoding does not seem to be consistent, so try variants
-  header = c()
-  Nlines = 16
-  while (length(header) == 0 | length(grep("Scale-3",header)) == 0 |
+  header = header_rownames = c()
+  Nlines = 18
+  options(warn = -1)
+  while (length(header) == 0 | 
+         length(grep("Scale-3",header)) == 0 |
          length(grep("Scale-2",header)) == 0 |
          length(grep("Scale-1",header)) == 0) { # as we do not know what header size is, search for it (needed in R version =< 3.1)
-
-    try(expr = {header = rownames(read.csv(filename, nrow = Nlines, header = TRUE))}, silent = TRUE)
+    
+    try(expr = {header = read.csv(filename, nrow = Nlines, header = TRUE)}, silent = TRUE)
     if (length(header) == 0) {
-      header = rownames(read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "WINDOWS-1252"))
+      header = read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "WINDOWS-1252")
     }
     if (length(header) == 0) {
-      header = rownames(read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "UTF-8"))
+      header = read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "UTF-8")
     }
     if (length(header) == 0) {
-      header = rownames(read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "latin1"))
+      header = read.csv(filename, skipNul = TRUE, nrow = Nlines, header = TRUE, fileEncoding = "latin1")
+    }
+    if (length(header) > 0) {
+      header_rownames = rownames(header)
+      if (isTRUE(all.equal(as.character(1:Nlines), header_rownames))) {
+        header = as.character(header[,1])
+      } else {
+        header = header_rownames
+      }
     }
     Nlines = Nlines - 1
-    if (Nlines == 10) print("Error: wav file header not recognized in function g.wavread")
+    if (Nlines == 10) print("Error: wav file header not recognized in function readWav from the GGIRread package")
   }
-  scale3position = grep("Scale-3", header)
+  if (length(grep("Scale-3",header)) > 0) scale3position = grep("Scale-3", header)
```

Comment:
Did you observe cases where "Scale-3" was not present in the header?
Note that the while statement above already checks for "Scale-3" to be present. So, I think this if-statement is not needed.

Reply:
No, not really, I was not familiar with the wav heading and in my first attempts I did not realize that the scale-3 was always available.